### PR TITLE
bump golang.org/x/net to 0.17 to fix CVE-2023-39325

### PR DESCRIPTION
**Pull request by joelsmith**
Fix [GO-2023-2102](https://pkg.go.dev/vuln/GO-2023-2102) / [CVE-2023-39325](https://www.cve.org/CVERecord?id=CVE-2023-39325)
```
$ go get golang.org/x/net@v0.17.0
$ go mod tidy
$ go mod vendor
```

Before:

```
$ make adapter
...
$ govulncheck  --mode=binary bin/keda-adapter 
Scanning your binary for known vulnerabilities...

Vulnerability #1: GO-2023-2102
    HTTP/2 rapid reset can cause excessive work in net/http
  More info: https://pkg.go.dev/vuln/GO-2023-2102
  Module: golang.org/x/net
    Found in: golang.org/x/net@v0.15.0
    Fixed in: golang.org/x/net@v0.17.0
    Example traces found:
      #1: http2.Server.ServeConn

Your code is affected by 1 vulnerability from 1 module.

Share feedback at https://go.dev/s/govulncheck-feedback.
```

After:

```
$ make adapter
...
$ govulncheck  --mode=binary bin/keda-adapter 
Scanning your binary for known vulnerabilities...

No vulnerabilities found.

Share feedback at https://go.dev/s/govulncheck-feedback.
```

### Checklist

- [X] Changelog has been updated and is aligned with our [changelog requirements](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#Changelog)
- [X] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))

### CODE REVIEW

**Review by JorTurFer — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -70,6 +70,7 @@ Here is an overview of all new **experimental** features:
 ### Fixes
 
 - **General**: Prevented stuck status due to timeouts during scalers generation ([#5083](https://github.com/kedacore/keda/issues/5083))
+- **General**: Fix CVE-2023-39325 in golang.org/x/net ([#5121](https://github.com/kedacore/keda/issues/5121))
```
Comment (JorTurFer) [inline CHANGELOG.md:73]:
```suggestion
- **General**: Fix CVE-2023-39325 in golang.org/x/net ([#5121](https://github.com/kedacore/keda/issues/5121))
- **General**: Prevented stuck status due to timeouts during scalers generation ([#5083](https://github.com/kedacore/keda/issues/5083))
```

Comment (joelsmith) [reply CHANGELOG.md]: Fixed, thanks

**Review by zroubalik — APPROVED** [review]

**Pull request merged**